### ISSUE TITLE
theme installation/unistallation problem

### ISSUE THREAD

**Issue by Gabri** [quickapps/cms#24]
I encounter (again) problems on installation of a theme (zip made with the shell script).
After the installation I cannot see the default image of the theme and the information contained in the yaml file (the 'settings' field of 'module' table is NULL).
Then if I try to uninstall the theme I have an error message at the top of the page:
"Error uninstalling theme '[Theme name]'"
So I checked the error.log file. I discovered that during these opeations (install/uninstall) I have this stack trace:

2012-01-22 15:03:12 Error: [NotFoundException] The requested file was not found
#0 /home/gabri/workspace/uacepquickapp/app/Cake/Controller/Controller.php(933): MediaView->render(NULL, NULL)
#1 /home/gabri/workspace/uacepquickapp/app/Cake/Routing/Dispatcher.php(111): Controller->render()
#2 /home/gabri/workspace/uacepquickapp/app/Cake/Routing/Dispatcher.php(86): Dispatcher->_invoke(Object(ThemesController), Object(CakeRequest), Object(CakeResponse))
#3 /home/gabri/workspace/uacepquickapp/webroot/index.php(47): Dispatcher->dispatch(Object(CakeRequest), Object(CakeResponse))
#4 {main}

I have this stack trace twice for every operation (with same text and timestamp too)

Initially I tought that the problem could be the fact that I used (in the first test) a Theme name with a quote character inside; anyway trying to make a TestTheme with no strage chars and I have the same behavior.

I hope it helps

**Comment by Gabri** [comment]
Maybe the stack trace isn't related to the problem.
Clicking on 'set as default' link the operation works fine (I tried with 2 different themes and the Default too) but the stack traces were always shown in these operations too

**Comment by quickapps** [comment]
I was unable to reproduce this issue. (fresh new install from github package)
Shell seems to work fine, Ive created a new theme and:
- installation, ok
- set as default, ok
- theme's thumbnail image, shows ok
- uninstall, ok
- "the 'settings' field of 'module' table is NULL": This is completely normal, if you have not configured your theme after installation there are not settings.
- "NotFoundException": this is may be related to the missing theme's thumbnail

---

Can you check this?:
Check the method `admin_theme_tn` in controller: `Plugin/System/Controller/ThemesController.php`
And try to find the cause of the `thumbnail issue`. Because I was unable to reproduce this issue.

Also, after theme installation clear all your cache files on `/tmp/cache`

I think that all this may be related to some enviroment factors.
Anyway, I'll keep trying to reproduce this, and find the cause.

## 

Thanks!
Chris

**Comment by Gabri** [comment]
Found the problem!  :)

The issue regards the Theme name!
During the theme production, shell sais:
"Alias name of the theme, in CamelCase. e.g.: "MyTestTheme" [R]"
And every time I typed something like "DummyTheme" or "TestTheme"
Everything works fine if I don't use the suffix "Theme".
In wiki I haven't found that is not possible to use that suffix, but probably it could be good to check on shell script this kind of constraint (if it is) and ask to type again a new name.

In my case: 
Theme name: TestTheme
the function admin_theme_tn produced a path like this: "/home/gabri/workspace/uacepquickapp/Themes/Themed/Test/"
and not "/home/gabri/workspace/uacepquickapp/Themes/Themed/TestTheme/"

I think the uninstallation issue has a similar problem even if I put the suffix "Theme" on the unistallation url it raises the same message.
